src/Microsoft.AspNet.Mvc.Core/Rendering/ModelExpression.cs

```diff
@@ -35,10 +33,24 @@ public ModelExpression([NotNull] string name, [NotNull] ModelMetadata metadata)
         /// <summary>
         /// Metadata about the <see cref="System.Linq.Expressions.Expression"/> of interest.
         /// </summary>
+        public ModelMetadata Metadata { get { return ModelExplorer.Metadata; } }
+
+        /// <summary>
+        /// Gets the model object for the <see cref="System.Linq.Expressions.Expression"/> of interest.
+        /// </summary>
+        /// <remarks>
+        /// Getting <see cref="Model"/> will evaluate a compiled version of the original
+        /// <see cref="System.Linq.Expressions.Expression"/>.
+        /// </remarks>
+        public object Model { get { return ModelExplorer.Model; } }
```

Comment:
I don't really agree, I think you're generally **most** interested in `Model` and/or `Metadata`.